Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1062 @@
+
+    #$requiredMinTLSVersion = 'TLS10'
+
+    $validFrontDoorEndpointDetails | ForEach-Object {
+        $frontdoorEndpoint = $_
+        $resourceGroupName = $_.ResourceGroupName
+        $resourceName = $_.ResourceName
+        $DomainName = $_.DomainName
+        $CertificateType =$_.CertificateType
+        $minimumTlsVersion = $_.MinimumTlsVersion
+        $prevMinimumTlsVersion =$_.PrevMinimumTlsVersion
+        $isMinTLSVersionSetOnCustomDomain = $_.isMinTLSVersionSetOnCustomDomain
+        $isMinTLSVersionRolledBackCustomDomainPostRemediation = $false;
+
+        $frontdoorEndpoint | Add-Member -NotePropertyName isMinTLSVersionRolledBackCustomDomainPostRemediation -NotePropertyValue $isMinTLSVersionRolledBackCustomDomainPostRemediation
+        
+        try
+        {
+            Write-Host "Fetching Front Door configuration: Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+
+                $secrets =  Get-AzFrontDoorCdnSecret -ResourceGroupName $resourceGroupName -ProfileName $resourceName
+                $secretObj = $null
+
+                foreach($secret in $secrets)
+                {
+                    if($secret.Name.Contains($DomainName))
+                    {
+                        $secretObj = $secret
+                    }
+                }
+                $secretResoure = New-AzFrontDoorCdnResourceReferenceObject -Id $secretObj.Id
+                $updateTlsSetting = New-AzFrontDoorCdnCustomDomainTlsSettingParametersObject -CertificateType $CertificateType -MinimumTlsVersion $prevMinimumTlsVersion -Secret $secretResoure
+                $resource = Update-AzFrontDoorCdnCustomDomain -ResourceGroupName $resourceGroupName -ProfileName $resourceName -CustomDomainName $DomainName -TlsSetting $updateTlsSetting
+                    
+                if($null -ne $resource -and $updateTlsSetting.MinimumTlsVersion -eq 'TLS10' )
+                {
+                    $frontdoorEndpoint.isMinTLSVersionRolledBackCustomDomainPostRemediation = $true
+                    $frontdoorEndpoint.isMinTlsVersionSetOnCustomDomain = $false
+                    $frontdoorEndpoint.minimumTlsVersion = $requiredMinTLSVersion
+                    $frontDoorsRolledBack += $frontdoorEndpoint
+                    Write-Host "Minimum required TLS version for Front Door has been Rolled back successfully for FrontDoor" ($resourceName) "with Domain" ($DomainName) -ForegroundColor $([Constants]::MessageType.Update)
+                    Write-Host $([Constants]::SingleDashLine)
+                }
+                else
+                {
+                    $frontDoorsEPSkipped += $frontdoorEndpoint
+                    $logResource = @{}
+                    $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                    $logResource.Add("ResourceName",($_.ResourceName))
```

Comment:
Plz check here if entire front door resource or front door endpoint will be skipped.

Reply:
Added domain name in logs